gobblin-compaction/src/main/java/org/apache/gobblin/compaction/mapreduce/orc/OrcUtils.java

```diff
@@ -491,18 +490,38 @@ public static WritableComparable createValueRecursively(TypeDescription schema)
    */
   public static boolean eligibleForUpConvertHelper(TypeDescription originalSchema, TypeDescription targetSchema) {
     if (!targetSchema.getCategory().isPrimitive()) {
-      if (!originalSchema.getFieldNames().containsAll(targetSchema.getFieldNames())) {
-        return false;
-      }
-      boolean result = true;
+      if (targetSchema.getCategory().equals(TypeDescription.Category.LIST)) {
+        Preconditions
+            .checkArgument(originalSchema.getChildren() != null, "Illegal format of ORC schema as:" + targetSchema);
+        return eligibleForUpConvertHelper(originalSchema.getChildren().get(0), targetSchema.getChildren().get(0));
+      } else if (targetSchema.getCategory().equals(TypeDescription.Category.MAP)) {
+        Preconditions
```

Comment:
Should the error message be: "Illegal format of ORC schema as:" + originalSchema?